packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+      _sell(tokenAddress);
+    }
+    _burnCelo();
+    _distribute(tokenAddress);
+    _distribute(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+  }
+
+  /**
+    * @notice Burns all the Celo balance of this contract.
+    */
+  function _burnCelo() private {
+    TokenState storage tokenState = tokenStates[registry.getAddressForOrDie(
+      GOLD_TOKEN_REGISTRY_ID
+    )];
+    ICeloToken celo = ICeloToken(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+
+    uint256 balanceOfCelo = celo.balanceOf(address(this));
+    uint256 balanceToProcess = balanceOfCelo.sub(tokenState.toDistribute);
+    uint256 balanceToBurn = FixidityLib
+      .newFixed(balanceToProcess)
+      .multiply(burnFraction)
+      .fromFixed();
+    celo.burn(balanceToBurn);
+
+    tokenState.toDistribute += balanceToProcess - balanceToBurn;
+  }
+
+  /**
+    * @param token The address of the token to query.
+    * @return The amount burned for a token.
+    */
+  function getPastBurnForToken(address token) external view returns (uint256) {
+    return tokenStates[token].pastBurn;
+  }
+
+  /**
+    * @param token The address of the token to query.
+    * @param amountToBurn The amount of the token to burn.
+    * @return Returns true if burning amountToBurn would exceed the daily limit.
+    */
+  function dailyBurnLimitHit(address token, uint256 amountToBurn) public returns (bool) {
+    TokenState storage tokenState = tokenStates[token];
+
+    if (tokenState.dailyBurnLimit == 0) {
+      // if no limit set, assume uncapped
+      return false;
+    }
+
+    uint256 currentDay = now / 1 days;
+    // Pattern borrowed from Reserve.sol
+    if (currentDay > lastLimitDay) {
+      lastLimitDay = currentDay;
+      tokenState.currentDaySellLimit = tokenState.dailyBurnLimit;
+    }
+
+    return amountToBurn >= tokenState.currentDaySellLimit;
+  }
+
+  /**
+    * @notice Updates the current day limit for a token.
+    * @param token The address of the token to query.
+    * @param amountBurned the amount of the token that was burned.
+    */
+  function updateLimits(address token, uint256 amountBurned) private {
+    TokenState storage tokenState = tokenStates[token];
+
+    if (tokenState.dailyBurnLimit == 0) {
+      // if no limit set, assume uncapped
+      return;
+    }
+    tokenState.currentDaySellLimit = tokenState.currentDaySellLimit.sub(amountBurned);
+    emit DailyLimitUpdated(amountBurned);
+    return;
+  }
+
+  /**
+    * @notice Allows owner to transfer tokens of this contract. It's meant for governance to 
+      trigger use cases not contemplated in this contract.
+      @param token The address of the token to transfer.
+      @param recipient The address of the recipient to transfer the tokens to.
+      @param value The amount of tokens to transfer.
+      @return A boolean indicating whether the transfer was successful or not.
+    */
+  function transfer(address token, address recipient, uint256 value)
```

Comment:
This seems extremely powerfull. Since it can transfer any balance even the part that was supposed to be burned. Should we limit this to only allow transfer of toDistribute balance?